client/src/app/site/agenda/components/topic-detail/topic-detail.component.ts

```diff
@@ -0,0 +1,171 @@
+            // after creating a new topic, go "back" to agenda list view
+            this.location.replaceState('/agenda/');
+        } else {
+            await this.repo.update(this.topicForm.value, this.topic);
+            this.setEditMode(false);
+        }
+    }
+
+    /**
+     * Setup the form to create or alter the topic
+     */
+    public createForm(): void {
+        this.topicForm = this.formBuilder.group({
+            title: ['', Validators.required],
+            text: ['']
+        });
+    }
+
+    /**
+     * Overwrite form Values with values from the topic
+     */
+    public patchForm(): void {
+        const topicPatch = {};
+        Object.keys(this.topicForm.controls).forEach(ctrl => {
+            topicPatch[ctrl] = this.topic[ctrl];
+        });
+        this.topicForm.patchValue(topicPatch);
+    }
+
+    /**
+     * Determine whether a new topic should be created or an existing one should
+     * be loaded using the ID from the URL
+     */
+    public getTopicByUrl(): void {
+        if (this.route.snapshot.url[1] && this.route.snapshot.url[1].path === 'new') {
+            // creates a new topic
+            this.newTopic = true;
+            this.editTopic = true;
+            this.topic = new ViewTopic();
+        } else {
+            // load existing topic
+            this.route.params.subscribe(params => {
+                this.loadTopic(params.id);
+            });
+        }
+    }
+
+    /**
+     * Loads a top from the repository
+     * @param id the id of the required topic
+     */
+    public loadTopic(id: number): void {
+        this.repo.getViewModelObservable(id).subscribe(newViewTopic => {
+            // repo sometimes delivers undefined values
+            // also ensures edition cannot be interrupted by autoupdate
+            if (newViewTopic && !this.editTopic) {
+                this.topic = newViewTopic;
+                // personalInfoForm is undefined during 'new' and directly after reloading
+                if (this.topicForm) {
+                    this.patchForm();
+                }
+            }
+        });
+    }
+
+    /**
+     * Create the absolute path to the corresponding list of speakers
+     */
+    public getSpeakerLink(): string {
+        if (!this.newTopic && this.topic) {
+            const item = this.repo.getAgendaItem(this.topic.topic);
+            if (item) {
+                return `/agenda/${item.id}/speakers`;
+            }
+        }
+    }
+
+    /**
+     * Handler for the delete button. Uses the PromptService
+     */
+    public async onDeleteButton(): Promise<any> {
+        const content = this.translate.instant('Delete') + ` ${this.topic.title}?`;
+        if (await this.promptService.open('Are you sure?', content)) {
+            await this.repo.delete(this.topic);
+            this.router.navigate(['/agenda']);
+        }
+    }
+
+    /**
+     * Hitting escape while in topicForm should cancel editing
```

Comment:
the `@param` is missing in documentation

Reply:
yep